src/cmd/tag.go

```diff
@@ -0,0 +1,235 @@
+}
+
+type TaggableResourceFetchAliasFunction func(alias string) (any, error)
+
+var TaggableResourceFetchAliasFunctions = map[opslevel.TaggableResource]TaggableResourceFetchAliasFunction{
+	opslevel.TaggableResourceService:                func(alias string) (any, error) { return getClientGQL().GetServiceWithAlias(alias) },
+	opslevel.TaggableResourceRepository:             func(alias string) (any, error) { return getClientGQL().GetRepositoryWithAlias(alias) },
+	opslevel.TaggableResourceTeam:                   func(alias string) (any, error) { return getClientGQL().GetTeamWithAlias(alias) },
+	opslevel.TaggableResourceUser:                   func(alias string) (any, error) { return getClientGQL().GetUser(alias) },
+	opslevel.TaggableResourceDomain:                 func(alias string) (any, error) { return getClientGQL().GetDomain(alias) },
+	opslevel.TaggableResourceSystem:                 func(alias string) (any, error) { return getClientGQL().GetSystem(alias) },
+	opslevel.TaggableResourceInfrastructureresource: func(alias string) (any, error) { return getClientGQL().GetInfrastructure(alias) },
+}
+
+func GetTags(obj interface{}) (*opslevel.TagConnection, error) {
+	// call Elem because input is a ptr
+	v := reflect.ValueOf(obj).Elem()
+	tagsField := v.FieldByName("Tags")
+
+	if tagsField.IsValid() {
+		return tagsField.Interface().(*opslevel.TagConnection), nil
+	} else {
+		return nil, errors.New("reflection error")
+	}
+}
+
+var resourceType string
+
+var createTagCmd = &cobra.Command{
+	Use:   "tag --type=RESOURCE_TYPE [--assign] RESOURCE_ID KEY VALUE",
+	Short: "Create/assign a tag",
+	Long:  "Create/assign a tag",
+	Example: `
+opslevel create tag --type=Team ID|ALIAS KEY VALUE
+`,
+	Args:       cobra.ExactArgs(3),
+	ArgAliases: []string{"RESOURCE_ID", "KEY", "VALUE"},
+	Run: func(cmd *cobra.Command, args []string) {
+		err := validateResourceTypeArg()
+		cobra.CheckErr(err)
+
+		resource := args[0]
+		key := args[1]
+		value := args[2]
+
+		if cmd.Flag("assign").Changed {
+			tagInput := opslevel.TagInput{
+				Key:   key,
+				Value: value,
+			}
+			input := opslevel.TagAssignInput{Tags: []opslevel.TagInput{tagInput}}
+
+			if opslevel.IsID(resource) {
+				input.Id = opslevel.ID(resource)
+			} else {
+				input.Alias = resource
+				input.Type = opslevel.TaggableResource(resourceType)
+			}
+
+			result, err := getClientGQL().AssignTag(input)
+			cobra.CheckErr(err)
+			fmt.Printf("updated new tag on %s: %s\n", resource, result[0].Id)
+		} else {
+			input := opslevel.TagCreateInput{
+				Key:   key,
+				Value: value,
+			}
+			if opslevel.IsID(resource) {
+				input.Id = opslevel.ID(resource)
+			} else {
+				input.Alias = resource
+				input.Type = opslevel.TaggableResource(resourceType)
+			}
+
+			result, err := getClientGQL().CreateTag(input)
+			cobra.CheckErr(err)
+			fmt.Printf("added new tag on %s: %s\n", resource, result.Id)
```

Comment:
```suggestion
			fmt.Printf("created tag for %s: %s\n", resource, result.Id)
```

Mirrors the wording for `create tag` and `assign tag` just so that its a bit more clear